src/toast/styled-components.js

```diff
@@ -82,12 +82,12 @@ export const Root = styled('div', (props: ToasterSharedStylePropsT) => {
 });
 
 export const Body = styled('div', (props: SharedStylePropsT) => {
-  const {$isHidden, $isAnimating, $kind, $theme} = props;
+  const {$isVisible, $kind, $theme} = props;
   return {
     ...$theme.typography.font300,
     pointerEvents: 'auto',
     color: $theme.colors.white,
-    height: $isHidden && !$isAnimating ? 0 : 'auto',
+    height: 'auto',
```

Comment:
Animating between `0` and `auto` doesn't actually work, you can only animate between two literal values., so this wasn't actually do anything.

Reply:
but `opacity` animation is different from `height`. Which one is supposed to be? I think, for toast it has to be `height`, so should you try '100%' instead of `auto`, but in this case we need to reconsider outer container of it

Reply:
`100%` would work, but as you pointed out you need to also have a container with a fixed height for that to work, so I'm not sure that's a viable solution here. 

If I recall correctly the only reiable way to animate an element from 0 height to its "auto" height is to first render it hidden, use `ref.current.clientHeight` or `ref.current.getBoundingClientRect().height`, and then make it visible and transition from 0 to that explicit height value.

Reply:
so we have to render it to know exact height. So some day I sued the next trick: I rendered it inside of fake container to get height and than used it in real DOM. Look at this below. It shouldn't take much of performance.

https://www.npmjs.com/package/react-rendered-size
https://github.com/planttheidea/react-rendered-size/blob/master/src/element.js#L16